### PR TITLE
Added some checks to ensure there is a requirements tree before walking it

### LINKED ISSUE #1222: AttributeError: 'NoneType' object has no attribute 'get_children'
Sentry Issue: [MITXONLINE-2F6](https://sentry.io/organizations/mit-office-of-digital-learning/issues/3745113895/?referrer=github_integration)

```
AttributeError: 'NoneType' object has no attribute 'get_children'
(14 additional frame(s) were not displayed)
...
  File "rest_framework/serializers.py", line 663, in to_representation
    return [
  File "rest_framework/serializers.py", line 664, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "rest_framework/serializers.py", line 515, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "rest_framework/fields.py", line 1882, in to_representation
    return method(value)
  File "courses/serializers.py", line 354, in get_requirements
    for node in req_root.get_children():
```

### PR DESCRIPTION
**Pull request by jkachel**
#### Pre-Flight checklist

- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

Fixes #1222 

#### What's this PR do?

Adds a check to see if there is a requirements tree for the program before generating the course ID list. (If the program record predated the requirements tree, there wouldn't be a root node and you'd get a 500 error.) 

#### How should this be manually tested?

1. Create a program.
2. Manually delete the requirements tree nodes associated with that program. (For local testing, I did `delete from courses_programrequirement where program_id = ?` where `?` is the program ID in a database shell.) 
3. Try to load the program via the API: `/api/programs/<program id>/` - it should work properly, with an empty set for the requirements list.

### CODE REVIEW

File: courses/serializers.py [hunk 1]
```diff
@@ -370,12 +377,16 @@ def get_requirements(self, instance):
                     )
 
             else:
-                formatted_reqs["electives"] = [
-                    req.course.id
-                    for req in node.get_children()
-                    .filter(node_type=models.ProgramRequirementNodeType.COURSE)
-                    .all()
-                ]
+                formatted_reqs["electives"] = (
```
Comment (annagav) [inline courses/serializers.py:380]: This check chunk of code is similar to both electives and required. Could we refactor it to take it above the if and else?

**Review by annagav — CHANGES_REQUESTED** [review]
Functionally works great! Just a suggestion to refactor

File: courses/serializers.py (hunk 1 above)
Comment (jkachel) [reply courses/serializers.py]: Pulled this all into `_get_nested_requirements` in https://github.com/mitodl/mitxonline/pull/1223/commits/2f44857a3c6db05c871e164098074d19bb680d19 (and added an additional check to make sure the node has a course ID)

**Review by annagav — APPROVED** [review]
👍

**Pull request merged**